includes/abstracts/abstract-wc-legacy-order.php

```diff
@@ -0,0 +1,621 @@
+	 * @param int|object|WC_Order $order Order to init.
+	 */
+	protected function init( $order ) {
+		_deprecated_function( 'init', '2.7', 'Logic moved to constructor' );
+		if ( is_numeric( $order ) ) {
+			$this->read( $order );
+		} elseif ( $order instanceof WC_Order ) {
+			$this->read( absint( $order->get_id() ) );
+		} elseif ( isset( $order->ID ) ) {
+			$this->read( absint( $order->ID ) );
+		}
+	}
+
+	/**
+	 * Gets an order from the database.
+	 * @deprecated 2.7
+	 * @param int $id (default: 0).
+	 * @return bool
+	 */
+	public function get_order( $id = 0 ) {
+		_deprecated_function( 'get_order', '2.7', 'read' );
+		if ( ! $id ) {
+			return false;
+		}
+		if ( $result = get_post( $id ) ) {
+			$this->populate( $result );
+			return true;
+		}
+		return false;
+	}
+
+	/**
+	 * Populates an order from the loaded post data.
+	 * @deprecated 2.7
+	 * @param mixed $result
+	 */
+	public function populate( $result ) {
+		_deprecated_function( 'populate', '2.7', 'read' );
+		$this->read( $result->ID );
+	}
+
+	/**
+	 * Cancel the order and restore the cart (before payment).
+	 * @deprecated 2.7.0 Moved to event handler.
+	 * @param string $note (default: '') Optional note to add.
+	 */
+	public function cancel_order( $note = '' ) {
+		_deprecated_function( 'cancel_order', '2.7', 'update_status' );
+		WC()->session->set( 'order_awaiting_payment', false );
+		$this->update_status( 'cancelled', $note );
+	}
+
+	/**
+	 * Record sales.
+	 * @deprecated 2.7.0
+	 */
+	public function record_product_sales() {
+		_deprecated_function( 'record_product_sales', '2.7', 'wc_update_total_sales_counts' );
+		wc_update_total_sales_counts( $this->get_id() );
+	}
+
+	/**
+	 * Increase applied coupon counts.
+	 * @deprecated 2.7.0
+	 */
+	public function increase_coupon_usage_counts() {
+		_deprecated_function( 'increase_coupon_usage_counts', '2.7', 'wc_update_coupon_usage_counts' );
+		wc_update_coupon_usage_counts( $this->get_id() );
+	}
+
+	/**
+	 * Decrease applied coupon counts.
+	 * @deprecated 2.7.0
+	 */
+	public function decrease_coupon_usage_counts() {
+		_deprecated_function( 'decrease_coupon_usage_counts', '2.7', 'wc_update_coupon_usage_counts' );
+		wc_update_coupon_usage_counts( $this->get_id() );
+	}
+
+	/**
+	 * Reduce stock levels for all line items in the order.
+	 * @deprecated 2.7.0
+	 */
+	public function reduce_order_stock() {
+		_deprecated_function( 'reduce_order_stock', '2.7', 'wc_reduce_stock_levels' );
+		wc_reduce_stock_levels( $this->get_id() );
+	}
+
+	/**
+	 * Send the stock notifications.
+	 * @deprecated 2.7.0 No longer needs to be called directly.
+	 */
+	public function send_stock_notifications( $product, $new_stock, $qty_ordered ) {
+		_deprecated_function( 'send_stock_notifications', '2.7' );
+	}
+
+	/**
+	 * Output items for display in html emails.
+	 * @deprecated 2.7.0 Moved to template functions.
+	 * @param array $args Items args.
+	 * @return string
+	 */
+	public function email_order_items_table( $args = array() ) {
```

Comment:
Should this have a `_deprecated_function()` to trigger a notice? Seems like a good idea to encourage the use of `wc_get_email_order_items()` instead of this method.